src/atomate2/abinit/jobs/base.py

```diff
@@ -0,0 +1,815 @@
+        if not self.dependencies:
+            return
+
+        if restart_from is None:
+            prev_outputs_job_types = tuple(
+                [prev_output.calc_type for prev_output in prev_outputs]
+                if prev_outputs
+                else []
+            )
+            for job_type, deps_list in self.dependencies.items():
+                n_prev_jobs = prev_outputs_job_types.count(job_type)
+                if n_prev_jobs == 0:
+                    msg = f'No previous job of type "{job_type}".'
+                    logger.error(msg)
+                    raise InitializationError(msg)
+                elif n_prev_jobs > 1:
+                    msg = f'More than 1 previous job of type "{job_type}". Risk of overwriting.'
+                    logger.warning(msg)
+                self.resolve_deps_per_job_type(
+                    [
+                        prev_output
+                        for prev_output in prev_outputs
+                        if prev_output.calc_type == job_type
+                    ],
+                    deps_list,
+                )
+        else:
+            # Just link everything from the indata folder of the previous run.
+            # Files needed for restart will be overwritten
+            prev_indata = os.path.join(restart_from.dir_name, INDIR_NAME)
+            for f in os.listdir(prev_indata):
+                # if the target is already a link, link to the source to avoid many nested levels of linking
+                source = os.path.join(prev_indata, f)
+                if os.path.islink(source):
+                    source = os.readlink(source)
+                os.symlink(source, os.path.join(self.workdir, INDIR_NAME, f))
+
+    def resolve_deps_per_job_type(self, prev_outputs, deps_list):
+        """Resolve dependencies for specific job type."""
+        deps_list = deps_list if isinstance(deps_list, list) else [deps_list]
+        for prev_output in prev_outputs:
+            for dep in deps_list:
+                # TODO: Do we need to keep this here as it is supposed to be passed using the jobflow db ?
+                #  this is related to the question on abinit_input AND struture passed together in make.
+                #  Do we keep this thing with '@' ?
+                # if dep.startswith("@structure"):
+                #     self.abinit_input.set_structure(structure=prev_output.structure)
+                # if not dep.startswith("@"):
+                source_dir = prev_output.dir_name
+                self.abinit_input_set.set_vars(irdvars_for_ext(dep))
+                if dep == "DDK":
+                    raise NotImplementedError
+                    # self.link_ddk(source_dir)
+                elif dep == "1WF" or dep == "1DEN":
+                    raise NotImplementedError
+                    # self.link_1ext(dep, source_dir)
+                else:
+                    self.link_ext(dep, source_dir)
+
+    def link_ext(self, ext, source_dir, strict=True):
```

Comment:
For example, this could be a standalone function in a different module. In fact, symlinking is a really useful feature, so ideally it could be refactored into something more general, along the lines of the other file operation functions: https://github.com/materialsproject/atomate2/blob/614ace0ee657759091b6e39296614b07b42140c8/src/atomate2/common/files.py#L21